docs(contributing): Write about Docker, Cube.js Server (#1806)
cube-js/cube.js: CONTRIBUTING.md

Cube.js works with Node.js 8+ and uses yarn as a package manager.

## Development Workflow

### Cube.js Docker

> Cube.js offers two different types of docker's image:

Reviewer: ```suggestion
Cube.js offers two different types of Docker image:
```

- Stable (building from published release on npm)
- Dev (building from source files, needed to test unpublished changes)

For more information, take a look at [Docker Development Guide](./packages/cubejs-docker/DEVELOPMENT.md).

#### Stable Docker Release

> 1. After cloning Cube.js repository run `$ yarn` in `packages/cubejs-docker` to install dependencies.

Reviewer: Can we remove the `$` from the beginning of all the inline commands, please?

Author: I am using similar style across file, it was not added by me. 

cC @keydunov Is it ok to remove `$`?

Thanks

Reviewer 2: Let's keep `$` for now as we're using it everywhere

2. Use `$ docker build -t cubejs/cube:latest -f latest.Dockerfile` in `packages/cubejs-docker` to build stable docker image.

#### Development

1. After cloning Cube.js repository run `$ yarn` and `$ yarn lerna bootstrap` to install dependencies.
2. Use `$ docker build -t cubejs/cube:dev -f dev.Dockerfile ../../` to build stable development image.

### Cube.js Client

1. After cloning Cube.js repository run `$ yarn install` and `$ yarn lerna bootstrap` in root directory.
2. Use `$ yarn link` to add these packages to link registry.
3. Perform required code changes.
4. Use `$ yarn build` in the repository root to build CommonJS and UMD modules.
5. Use `$ yarn link @cubejs-client/core` and/or `$ yarn link @cubejs-client/react` in your project to test changes applied.
6. Use `$ yarn test` where available to test your changes.

> 7. Ensure to commit CommonJS and UMD modules as part of your commit.

Reviewer: ```suggestion
7. Ensure that any CommonJS and UMD modules are included as part of your commit.
```

To get set up quickly, you can perform 1) and 2) with one line from the `cube.js` clone root folder:

```
cd packages/cubejs-client-core && yarn && yarn link && cd ../.. && cd packages/cubejs-client-react && yarn && yarn link && cd ../..
```

### Cube.js Server

> Cube.js is written in plain JS, but some parts already migrated to TypeScript.

Reviewer: ```suggestion
Cube.js is written in plain JavaScript, but some parts have already been migrated to TypeScript.
```
